pkg/filepromoter/gzip.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package filepromoter
+
+import (
+	"compress/gzip"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"k8s.io/klog"
+)
+
+// minCompressSize is the minimum size for even trying to compress files
+const minCompressSize = 8 * 1024
+
+// minCompressionRatio is the ratio we must obtain to consider a file compressible
+const minCompressionRatio = 0.9
```

Comment:
Can you explain the ration in plain English (maybe give an example)? It's not obvious if the 0.9 here means 90% reduction in size or a 10% reduction in size (or something else).